torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -11,90 +11,88 @@ namespace torch {
 namespace jit {
 namespace fuser {
 namespace cuda {
+namespace kir {
 
 namespace {
 
 //! Scan all primary expressions in the Kernel IR and build
-//! list of specialized nodes
-//!
-//! \note primary expressions are expressions which are not subexpressions
-//!   in a larger expression (things like ForLoop or IfThenElse are not
-//!   real expressions)
-//!
-class KernelIrScanner : private OptOutDispatch {
+//! lists of specialized nodes and other interesting information
+class KernelIrScanner : private kir::IrVisitor {
```

Comment:
There are a lot of changes in this class. Can these changes be separated out as an individual refactoring PR?

Reply:
This is actually an example of cascading changes: 
1. switching from OptOutDispatch to IrVisitor. 
2. we don't have (or build) the list of primary expressions. 

#2 is also a consequence of the IrVisitor, which doesn't dispatch to abstract base classes (Expr). So yes, the changes are substantial but the result is much simpler.

Reply:
OK. Looks good to me.